### PR TITLE
sdk: put AccountSharedData::set_data() behind feature="dev-context-only-utils"

### PR DESCRIPTION
**Pull request by alessandrod**
set_data() is deprecated in favor of set_data_from_slice().

### CODE REVIEW

**Comment by ryoqun** [comment]
hey, thanks for trying using dcow. :) seems ci is green. is this reviewable now?

**Comment by alessandrod** [comment]
> hey, thanks for trying using dcow. :) seems ci is green. is this reviewable now?

yep!

File: accounts-db/src/append_vec/test_utils.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ pub fn get_append_vec_path(path: &str) -> TempFile {
 pub fn create_test_account(sample: usize) -> (StoredMeta, AccountSharedData) {
     let data_len = sample % 256;
     let mut account = AccountSharedData::new(sample as u64, 0, &Pubkey::default());
-    account.set_data((0..data_len).map(|_| data_len as u8).collect());
+    account.set_data_from_slice(&vec![data_len as u8; data_len]);
```
Comment (ryoqun) [inline accounts-db/src/append_vec/test_utils.rs:41]: :heart:

**Review by ryoqun — APPROVED** [review]
lgtm

**Pull request merged**